lib/chef/provider/package/zypper.rb

```diff
@@ -70,7 +71,34 @@ def get_current_versions
         end
 
         def candidate_version
-          @candidate_version ||= package_name_array.each_with_index.map { |pkg, i| available_version(i) }
+          if source_files_exist?
+            unless uri_scheme?(new_resource.source) || ::File.exist?(new_resource.source)
+              @package_source_exists = false
+              return
+            end
+
+            logger.trace("#{new_resource} checking rpm status")
+            shell_out!("rpm", "-qp", "--queryformat", "%{NAME} %{VERSION}-%{RELEASE}\n", new_resource.source).stdout.each_line do |line|
+              case line
+              when /^(\S+)\s(\S+)$/
+                current_resource.package_name($1)
+                new_resource.version($2)
+                @candidate_version = $2
+              end
+            end
+
+          else
+            @candidate_version ||= package_name_array.each_with_index.map { |pkg, i| available_version(i) }
+          end
+        end
+
+        def uri_scheme?(str)
+          scheme = URI.split(str).first
+          return false unless scheme
+
+          %w{http https ftp file}.include?(scheme.downcase)
```

Comment:
How did you determine this list? There's a long list [here](https://en.opensuse.org/openSUSE:Libzypp_URIs), not all of which might work for us because we'll have to deal with passing them to rpm in this implementation but I'm curious why these 4.